plugins/zfs/zfs.go

```diff
@@ -42,8 +47,49 @@ func getTags(kstatPath string) map[string]string {
 			pools += "::"
 		}
 		pools += pool
+
+		if poolMetrics {
+			err := readPoolStats(poolDir, pool, acc)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	return map[string]string{"pools": pools}, nil
+}
+
+func readPoolStats(poolIoPath string, poolName string, acc plugins.Accumulator) error {
```

Comment:
change this function name, `readPoolStats` suggests that you are not adding metrics to the accumulator here. Maybe something like `gatherPoolStats`